### PR TITLE
feat: add comments to updates

### LINKED ISSUE #1156: Comments in Project Updates
Every time a Grantee post an update, a comment thread in the Forum will be displayed in the individual project view.

Nice to have: A comment count in the Project Update card on the Grant view

Doubts:
- would updates have a forum thread of their own apart from them being shown within the comments?
- would updates allow users to leave comments?

If there are no comments, don't show the icon/count

If updates were to allow for comments, this is how i'd show them on the card/item
Image

What happens if user edits an update?
Automated message from the governance bot saying that the update has been edited in the coments thread

Copy for forum topic:

Title Update #(number) for (grant title) Grant

Author: published address
Update status: On Time / Late
Project Health: On Track/At Risk/Off Track

[message]

View this project update on the Governance dApp(Link to the update)

Copy for bot post:

EDIT
This project update has been edited by the author. Please check the latest status on the Governance dApp(Link to the update)

DELETE
This project update has been deleted by the author.

### PR DESCRIPTION
**Pull request by andyesp**
Closes #1156

### CODE REVIEW

File: src/back/services/update.tsx [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+  static async updateWithDiscoursePost(id: UpdateAttributes['id'], discoursePost: DiscoursePost) {
+    return await UpdateModel.update(
+      { discourse_topic_id: discoursePost.topic_id, discourse_topic_slug: discoursePost.topic_slug },
+      { id }
+    )
+  }
+
+  static commentUpdateEditInDiscourse(update: UpdateAttributes) {
+    console.log('uuuu', update)
```
Comment (ncomerci) [inline src/back/services/update.tsx:27]: image

**Review by ncomerci — APPROVED** [review]

File: src/components/Comments/Comments.tsx [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+  const [showAllComments, setShowAllComments] = useState(true)
+
+  useEffect(() => {
+    if (comments && comments.totalComments > DEFAULT_SHOWN_COMMENTS) {
+      setShowAllComments(false)
+    }
+  }, [comments])
+
+  const renderedComments = useMemo(() => {
+    if (comments && comments.totalComments > 0) {
```
Comment (1emu) [inline src/components/Comments/Comments.tsx:38]: this `comments && comments.totalComments > 0` can be `renderComments`

File: src/entities/Snapshot/utils.ts [hunk 3]
```diff
@@ -194,6 +193,7 @@ export function getChecksumAddress(address: string) {
   return ethers.utils.getAddress(address.toLowerCase())
 }
 
+// TODO: Move to non-snapshot-related file
```
Comment (1emu) [inline src/entities/Snapshot/utils.ts:196]: reminder
Could be user/utils

**Review by 1emu — APPROVED** [review]

File: src/back/routes/update.ts [hunk 4]
```diff
@@ -159,8 +185,14 @@ async function updateProposalUpdate(req: WithAuth<Request<{ proposal: string }>>
...
 
-  if (!completion_date) {
-    DiscordService.newUpdate(proposal.id, proposal.title, update.id, user)
+  const updatedUpdate = await UpdateService.getById(id)
+  if (updatedUpdate) {
+    if (!completion_date) {
+      DiscourseService.createUpdate(updatedUpdate, proposal.title)
+      DiscordService.newUpdate(proposal.id, proposal.title, update.id, user)
+    } else {
+      UpdateService.commentUpdateEditInDiscourse(updatedUpdate)
```
[the remark this answers is not in the document]
Comment (andyesp) [reply src/back/routes/update.ts:194]: Great idea, let's do that in another PR.

File: src/back/services/update.tsx [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+  static commentUpdateEditInDiscourse(update: UpdateAttributes) {
+    inBackground(async () => {
+      if (!update.discourse_topic_id) {
+        logger.error('No discourse topic associated to this update', { id: update.id })
+        return
+      }
+
+      await Discourse.get().commentOnPost({
+        topic_id: update.discourse_topic_id,
+        raw: `This project update has been edited by the author. Please check the latest status on the [Governance dApp](${getUpdateUrl(
```
Comment (ncomerci) [inline src/back/services/update.tsx:35]: Every time an author edits the same update, is this comment posted on the forum?

Comment (andyesp) [reply src/back/services/update.tsx:35]: Yes!

**Pull request merged**